### PR TITLE
Jacob/spotlight carousel

### PR DESCRIPTION
**Pull request by jaredhanson11**
Jacob's spotlight carousel addition

### CODE REVIEW

File: website/css/index.css [hunk 1]
```diff
@@ -607,13 +607,32 @@ body {
...
     text-align: justify;
     font-size: .8em;
 }
 
 #spotlight > .spotlight-picture > img {
     width: 100%;
+    border-radius: 20px;
+}
+
+#dots {
```
Comment (jaredhanson11) [inline website/css/index.css]: Make the dots have the mouse be a hand pointer so people know they can click the dots

File: website/create_slides.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import json
+slide = {'imageSrc': None, 'desc': None}
+
+i = 1
+while i < 4:
```
Comment (jaredhanson11) [inline website/create_slides.py]: If you want to make this configurable based on how many spotlightN.txt you have in the txt folder, you can look up 
```python
import os 
os.listdir('./txt/') 
```
which gives a list of filenames in the txt directory. Then use for loop through that list and you can add any amount of spotlight.txt's. This means you'd have to make dots configurable based on number of spotlightN variables in `spotlight.json`. Also I think you might want to add another line to spotlight.json maybe that is 
```javascript
var numSpotlightStories = N;
```
and then that will allow you to know how many spotlightN vars you have defined. Let me know if makes sense/you want to do it.

File: website/index.html [hunk 3]
```diff
@@ -296,18 +297,12 @@
...
-                them to The John Ernest Foundation, which provided their conference fees
-                to help them reach their goals. This project was conceived 1 month ago.
-                Talk about adding value in a timely manner!  Conference was held at Oxford
-                University in mid-December, 2018.
+            <div class="spotlight-picture"><img id="spotlightImg" src='./img/spotlight1.png' /></div>
+            <div id="spotlight-description">Eric Sodomka (center right) and Okke Schrijvers (center left) are senior scientists at Facebook headquarters in Menlo Park. Eric spent countless hours to get four young Nigerian young academics visas, conference fees, and flights to come to Oxford to attend the 14th conference on web and internet economics, a conference on mechanism design. Okke (long standing John Ernest Fellow and member of TJEF’s College of Mentors) connected them to The John Ernest Foundation, which provided their conference fees to help them reach their goals. This project was conceived 1 month ago. Talk about adding value in a timely manner! Conference was held at Oxford University in mid-December, 2018.</div>
```
Comment (jaredhanson11) [inline website/index.html]: I wouldn't make this default to that story, because what happens when Ken changes spotlight1.txt to a different story he wants to feature. It wouldn't automatically reflect, only once the first dot was clicked or the slideshow reset it'd be the correct values in spotlight.json.

File: website/js/index.js [hunk 4]
```diff
@@ -4,6 +4,64 @@ function parallaxScroll(currentScroll) {
...
+    if (slideIndex == 3)
+      slideIndex = 0
+
+    changeSlide(Number(slideIndex) + 1)
+}
+
+function changeSlide(n) {
+  $('#spotlightImg').fadeOut(1000, function() {
+    var dots = $('.dot')
+    for (i = 0; i < dots.length; i++)
```
Comment (jaredhanson11) [inline website/js/index.js]: make these two lines be:
```javascript
dots.map((curr, index) => {
    curr.className = curr.className.replace(' active', '')
    if (index == n-1) {
        dots[n-1].className += ' active'    
    }
});
```

**Comment by jaredhanson11** [comment]
Just a general comment also. I think it looks bad with images that are different sizes having the description pop in and out of the page. I think we should standardize the height of all the images and make them fit the certain height or add padding if they're not a big enough image so that the description stays in relatively the same spot on the page when switching stories.

File: website/index.html [hunk 5]
```diff
@@ -6,6 +6,7 @@
         <title>TJEF</title>
         <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
         <script type="text/javascript" src="./js/index.js"></script>
+        <script type="text/javascript" src="./js/spotlight.json"></script>
```
[the remark this answers is not in the document]
Comment (jhanreg11) [reply website/index.html]: resolved

File: website/create_slides.py (hunk 2 above)
Comment (jhanreg11) [reply website/create_slides.py]: Resolved

File: website/index.html (hunk 3 above)
Comment (jhanreg11) [reply website/index.html]: resolved

File: website/index.html [hunk 6]
```diff
@@ -296,18 +297,12 @@
...
-                to help them reach their goals. This project was conceived 1 month ago.
-                Talk about adding value in a timely manner!  Conference was held at Oxford
-                University in mid-December, 2018.
+            <div class="spotlight-picture"><img id="spotlightImg" src='./img/spotlight1.png' /></div>
+            <div id="spotlight-description">Eric Sodomka (center right) and Okke Schrijvers (center left) are senior scientists at Facebook headquarters in Menlo Park. Eric spent countless hours to get four young Nigerian young academics visas, conference fees, and flights to come to Oxford to attend the 14th conference on web and internet economics, a conference on mechanism design. Okke (long standing John Ernest Fellow and member of TJEF’s College of Mentors) connected them to The John Ernest Foundation, which provided their conference fees to help them reach their goals. This project was conceived 1 month ago. Talk about adding value in a timely manner! Conference was held at Oxford University in mid-December, 2018.</div>
+            <div id="dots">
+              <span class="dot active"></span>
```
[the remark this answers is not in the document]
Comment (jhanreg11) [reply website/index.html]: resolved

File: website/css/index.css (hunk 1 above)
Comment (jhanreg11) [reply website/css/index.css]: resolved

File: website/js/index.js [hunk 7]
```diff
@@ -4,6 +4,64 @@ function parallaxScroll(currentScroll) {
...
+}
+
+function stopSlideShow() {
+  clearInterval(intervalId)
+}
+
+$(document).ready(function() {
+  startSlideShow()
+
+  $('.dot:first').click(function() {
```
[the remark this answers is not in the document]
Comment (jhanreg11) [reply website/js/index.js]: resolved

**Comment by jaredhanson11** [comment]
go ahead and close this PR and delete branch if it's not gonna be used/ you don't have any code from here you want to save for later.

**Comment by jaredhanson11** [comment]
Closing this PR as outdated. #8 added functionality this was going to add. @jhanreg11 please delete this branch if there is nothing else that you need from it.
